File: app/src/main/java/com/suzei/minote/ui/editor/EditorActivity.java
package com.suzei.minote.ui.editor;

import android.content.SharedPreferences;
import android.os.Bundle;

import com.suzei.minote.R;
import com.suzei.minote.data.DataSourceImpl;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.preference.PreferenceManager;

public class EditorActivity extends AppCompatActivity {

    public static final String EXTRA_NOTE_ID = "EXTRA_NOTE_ID";

    public static final String FRAGMENT_EDITOR_TAG = "EDITOR_FRAGMENT";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_editor);

        int itemId = getIntent().getIntExtra(EXTRA_NOTE_ID, -1);

        FragmentManager fm = getSupportFragmentManager();

        EditorFragment editorFragment = (EditorFragment) fm.findFragmentByTag(FRAGMENT_EDITOR_TAG);

        if (editorFragment == null) {

            editorFragment = EditorFragment.newInstance();

            fm.beginTransaction()
                    .replace(R.id.editor_container, editorFragment, FRAGMENT_EDITOR_TAG)
                    .commit();

            if (itemId != -1) {
                new EditorPresenter(itemId,
                        new DataSourceImpl(getApplicationContext()),
                        editorFragment);
            }
            else {
                SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
                new EditorPresenter(prefs,
                        new DataSourceImpl(getApplicationContext()),
                        editorFragment);
            }

        }

    }

}

File: app/src/main/java/com/suzei/minote/ui/editor/EditorActivity.kt
package com.suzei.minote.ui.editor

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.PreferenceManager
import com.suzei.minote.Injection
import com.suzei.minote.R

class EditorActivity : AppCompatActivity() {

    companion object {

        val EXTRA_NOTE_ID = "EXTRA_NOTE_ID"

        val FRAGMENT_EDITOR_TAG = "EDITOR_FRAGMENT"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_editor)

        val itemId = intent.getIntExtra(EXTRA_NOTE_ID, -1)

        val fm = supportFragmentManager

        var editorFragment = fm.findFragmentByTag(FRAGMENT_EDITOR_TAG) as EditorFragment?

        if (editorFragment == null) {

            editorFragment = EditorFragment.newInstance()

            fm.beginTransaction()
                    .replace(R.id.editor_container, editorFragment!!, FRAGMENT_EDITOR_TAG)
                    .commit()

            if (itemId != -1) {
                EditorPresenter(itemId,
                        Injection.provideDataSourceImpl(applicationContext),
                        editorFragment)
            } else {
                val prefs = PreferenceManager.getDefaultSharedPreferences(this)
                EditorPresenter(prefs,
                        Injection.provideDataSourceImpl(applicationContext),
                        editorFragment)
            }

        }

    }

}